lib/project_and_version_value.rb

```diff
@@ -1,55 +1,53 @@
-              where( "start_date IS NOT NULL AND due_date IS NOT NULL" ).
-              joins(:time_entries).
-              group("spent_on").collect { |issue| [issue.spent_on, issue.sum_hours] }
+  def project_costs(proj)
+    Issue.cross_project_scope(proj, 'descendants')
+      .select('MAX(spent_on) AS spent_on, SUM(hours) AS sum_hours')
+      .where('start_date IS NOT NULL AND due_date IS NOT NULL')
+      .joins(:time_entries)
+      .group('spent_on').collect { |issue| [issue.spent_on, issue.sum_hours] }
   end
 
-  def version_issues proj_id, version_id
+  def version_issues(proj_id, version_id)
     proj = Project.find(proj_id)
-    issues = Issue.cross_project_scope(proj, "descendants").
-              where( "start_date IS NOT NULL AND due_date IS NOT NULL AND fixed_version_id = ? ", version_id)
+    Issue.cross_project_scope(proj, 'descendants')
+      .where('start_date IS NOT NULL AND due_date IS NOT NULL AND fixed_version_id = ? ', version_id)
   end
 
-  def version_costs proj_id, version_id
+  def version_costs(proj_id, version_id)
     proj = Project.find(proj_id)
-    costs = Issue.cross_project_scope(proj, "descendants").
-              select("MAX(spent_on) AS spent_on, SUM(hours) AS sum_hours").
-              where( "start_date IS NOT NULL AND due_date IS NOT NULL AND fixed_version_id = ? ", version_id).
-              joins(:time_entries).
-              group("spent_on").collect { |issue| [issue.spent_on, issue.sum_hours] }
+    Issue.cross_project_scope(proj, 'descendants')
+      .select('MAX(spent_on) AS spent_on, SUM(hours) AS sum_hours')
+      .where('start_date IS NOT NULL AND due_date IS NOT NULL AND fixed_version_id = ? ', version_id)
+      .joins(:time_entries)
+      .group('spent_on').collect { |issue| [issue.spent_on, issue.sum_hours] }
   end
 
-  def incomplete_project_issues proj, basis_date
-    issues = Issue.cross_project_scope(proj, "descendants").
-              where( "start_date IS NOT NULL AND start_date <= ? AND due_date IS NOT NULL AND (closed_on IS NULL OR closed_on > ?)", basis_date, basis_date.to_time.end_of_day)
+  def incomplete_project_issues(proj, basis_date)
+    Issue.cross_project_scope(proj, 'descendants')
+      .where('start_date IS NOT NULL AND start_date <= ? AND due_date IS NOT NULL AND (closed_on IS NULL OR closed_on > ?)', basis_date, basis_date.to_time.end_of_day)
   end
 
-  def project_varsion_id_pair proj
-    issues = Issue.cross_project_scope(proj, "descendants").
-              where( "start_date IS NOT NULL AND due_date IS NOT NULL AND fixed_version_id IS NOT NULL" ).
-              joins(:fixed_version).order("effective_date ASC").
-              uniq.pluck(:project_id, :fixed_version_id)
+  def project_varsion_id_pair(proj)
+    Issue.cross_project_scope(proj, 'descendants')
+      .where('start_date IS NOT NULL AND due_date IS NOT NULL AND fixed_version_id IS NOT NULL')
+      .joins(:fixed_version).order('effective_date ASC')
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.